### ISSUE TITLE
Cursor ocassionally stops

### ISSUE THREAD

**Issue by Auruhausu** [ppy/osu#27227]
### Type

Game behaviour

### Bug description

After updating lazer I've discovered a really annoying thing: when playing STD (may also happen in Song Select menu) cursor ocassionaly stops while dragging tablet pen and after some time just "teleports" to the point where the pen is currently located. It happens anytime during gameplay and it is especially irritating when it happens at the end of a song.

### Screenshots or videos

https://github.com/ppy/osu/assets/152286401/b5bf8143-000a-4bee-975c-5d32c992b917

### Version

2024.131.0

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14323262/compressed-logs.zip)

**Comment by longnguyen2004** [comment]
Tracked at https://github.com/ppy/osu-framework/issues/6181

**Comment by frenzibyte** [comment]
Keeping this issue and pinning to give visibility for anyone else coming to report it.

**Comment by peppy** [comment]
Does turning off "high precision input mouse" with this?

![2024-02-19 11 12 26@2x](https://github.com/ppy/osu/assets/191335/d942d4b3-1c36-4cb6-b39d-03ab0f2f5e9e)

**Comment by Auruhausu** [comment]
Tried both turning off and on this option, but the problem still shows up

**Comment by BenCheung0422** [comment]
For my experience, I have a kind of 2 types of input lags: sudden stop and delayed movements, and luckily I am able to record them. For the type with sudden slow cursor movements, I have experienced for periods vary from a few hundreds of milliseconds to a few seconds. And sometimes coming with rendering lags, but it seems like it would be another problem.

- Sudden stop (blue circles "1" to "2"):

https://github.com/ppy/osu/assets/74168521/2b300a42-3f08-4ba0-960d-56b6b50129de

- Sudden delayed movements (near the end of this play, and apparent drop in input FPS is recorded):

https://github.com/ppy/osu/assets/74168521/ec2139e2-7618-4fcd-a333-41889e869ea4

From #9283 (the case with rendering lag), it is mentioned that it might be GC-related, but for my recordings, no apparent relationships to GC can be found. In the mean time, I will still try to collect more information related to the lags and delays as useful as possible.

**Comment by BenCheung0422** [comment]
This time. I have tracked on the logs and probably found that it might be related to the slow frames on the threads.

https://github.com/ppy/osu/assets/74168521/ced6979b-bc86-4eff-8863-21f37c7a892e

Here, I used 2880x1620 DSR for my 1920x1080 monitor, and 8 font size instead of 12 (default) for the terminals.

**Comment by peppy** [comment]
Having the stats open probably does that, and is probably not related. 

Also having windows displayed above the game, and potentially the game not focused, will not help.

ALso it's not running in fullscreen mode since windows are being displayed.

**Comment by BenCheung0422** [comment]
I am not sure if stats affect that. Also, in the video clip, I used OBS so that the console windows on the back can be rendered in front of the game window. Yesterday, I just captured the moments with cursor movement pauses. I will try to record with stats opened and see if I can capture the moments with slow movements. It is like there is frame loss in input threads in this case. I will open a new discussion for this.

**Comment by iwa** [comment]
i had a few input stutters in the last 2-3 weeks, and recorded one tonight

here's the clip:
https://github.com/ppy/osu/assets/19956672/a9b5bb30-3953-48df-b4ec-1eefaccdab6d

the logs:
[compressed-logs.zip](https://github.com/user-attachments/files/15597191/compressed-logs.zip)

specs (dunno if you have them in the logs but in case):
- ryzen 7 7700x, rx 7800 xt, 32gb ddr5 ram
- wacom ctl-472, using external opentabletdriver
- in-game tablet support disabled, high precision mouse disabled (always had issue with this setting on when using my tablet)

if you need further details don't hesitate to @ me

**Comment by BenCheung0422** [comment]
@iwa I think they are still working on it to fix this issue.

**Comment by iwa** [comment]
> @iwa I think they are still working on it to fix this issue.

ikik but as it seems to be kind of an awkward issue, the more logs / cases they can analyze, the merrier i guess?

**Comment by Packsolite** [comment]
What i find interesting with this issue is that i'm playing lazer for years but it only started occuring around feburary this year.
Also you can easily recreate this by stressing your cpu using prime95.

**Comment by bdach** [comment]
> Also you can easily recreate this by stressing your cpu using prime95.

That sounds like conflating two issues with different causes unless prime95 is doing windows input hooks for _whatever_ reason which I find unlikely.

**Comment by Packsolite** [comment]
> > Also you can easily recreate this by stressing your cpu using prime95.
> 
> That sounds like conflating two issues with different causes unless prime95 is doing windows input hooks for _whatever_ reason which I find unlikely.

Not sure, could be some weird windows services, update or whatever program in the background causing this.
Stable/old client doesn't suffer from this issues when running stress tests (except a few dropped frames, but not like 1 sec input freezes).

**Comment by BenCheung0422** [comment]
In my opinion, I would think this might not be really related. In my observation, there are mainly 4 types of threads: input, audio, update, draw. Certainly, draw (rendering) would be mainly handled by GPU threads alone, either single or multi-thread, and both input, audio and update handled mainly by CPU process. Lagging in threads absolutely can have several causes, including but not limited to: background services, process bottlenecks, library/API (including Windows API) incompatibility, driver incompatibility, driver or service errors, incompatible BIOS configurations, etc. In my case, there is only a hanging in input thread is the cause of input lagging, as what is handled by the coming changes.

It seems like their believe is the issue related to some incompatibilities of Windows API input handling. Also, certainly, stressing CPU can cause all the threads (and processes) lagging as mostly they are handled by CPU, doubtlessly, so it might not be really related.

**Comment by iwa** [comment]
something is definitely wrong on the rendering side

I was having the freeze issue with Valorant so I disabled GPU Scheduling (windows 11), and my issue worsen drastically, to the point where my tablet would sometimes freeze and dont respond at all unless I restarted opentabletdriver
it never happened with stable, and nothing changed expect the GPU scheduling setting
I then tried @yomikomanda 's suggestion to set render mode to single thread and it fixed everything (played all day and no stutter nor tablet freeze)

I can retrieve clips & logs if needed

**Comment by bdach** [comment]
@iwa unless you can produce a video with frame graph showing (ctrl-f11) any speculation that this issue has anything "to do with the rendering side" is unsubstantiated. i don't even know if you're trying to report the same issue which the OP is.

**Comment by iwa** [comment]
> @iwa unless you can produce a video with frame graph showing (ctrl-f11) any speculation that this issue has anything "to do with the rendering side" is unsubstantiated. i don't even know if you're trying to report the same issue which the OP is.

fair enough, I should be able to provide a video proof sometime during this week
should i enable/grab something else during my testing besides frame graph + logs?

**Comment by BenCheung0422** [comment]
For your information, this case we would have this kind of situation or otherwise it should be another issue. Here, you can see that in the graph (by clicking twice CTRL-F11), you can see there are tall lines in ticking thread graph but not any observable and obvious tall lines in rendering thread graph. If you case is different (like rendering lags), that should be another issue.

![312904247-40662e8a-6eba-42be-8e60-49a2eedcedf6 - frame at 0m7s](https://github.com/ppy/osu/assets/74168521/e513317d-4dd0-48ba-a805-a75718f699ce)

**Comment by iwa** [comment]
alright, thanks for your clarifications

after reviewing the original clip i posted, i can understand that y'all feel like it's a different issue (clip is cutting during the stutter for whatever reasons), but i'm pretty confident that i get stutter like OP does, or at least it feels the same 
I'll change my recording method to be sure we can visualize the issue + include frame graph

the only thing that makes me think it's rendering-related is that, strangely enough, input stutters totally stopped when i switched to Single thread rendering mode, but without any proofs my thinking is pointless, so i'll work on that

**Comment by LWapsyTN** [comment]
i might have an idea what is causing this.
if you have a gamepad or controller connected disconnect it, if you also have any input mapping software such as rewasd or xoutput stop it, for some reason they cause this, i think it is part of the input mapping process, but if you have any of these running and experience this issue, stop them, even steam input, it might cause the same issue. 
this might not actually be the real reason, but it is worth a try, if might work for you same as it did to me.

**Comment by BenCheung0422** [comment]
I have none of these software running, but seemingly it would interact with windows input and could cause delays in osu input handling, as osu at the moment is not handling with really raw input signals/events.

**Comment by iwa** [comment]
i have never used this kind of software, always disabled steam input, and using a controller only when playing RL, controller is disconnected otherwise

i don't fully exclude steam hypothesis tho, will try to test with steam opened and steam input enabled

i was thinking about it but couldn't it be a regression with old lazer installations? something that maybe would "magically" get fixed in the local config by updating the single/multi threaded setting?
some of my friends installed lazer a month ago, some are playing with a tablet with external opentabletdriver, and they never had any kind 
_(100% speculation here, idk how lazer works internally, trying to figure out as much ideas as i can)_

**Comment by ni5arga** [comment]
Had the same issue, tried switching from multi thread rendering to single thread rendering and it seems to be fine now.

**Comment by iwa** [comment]
@ni5arga if you switch back to multi threaded, does the issue reappears for you by any chance?

**Comment by BenCheung0422** [comment]
It does not help anything for my case. Logically, rendering is tacked by renderer (either GPU or CPU), but input is tacked by CPU, so there seems to be no obvious intersection between them. However, changing max FPS value may change something, but the input thread would still be 1000Hz (for me).

**Comment by BraianAzcune** [comment]
In the past two months, I've noticed that I'm also experiencing this issue while using the mouse (so far, it hasn't occurred when using the tablet). I've tried all the graphics engines—DirectX, OpenGL, Vulkan—and the issue persists regardless. I've also tested with and without enabling the high-precision mouse option, and even disconnected my tablet in case there was any interference, but the problem still occurs. I'm sharing two videos where you can see that the mouse suddenly freezes.

The first one is more subtle:In the past two months, I've noticed that I'm also experiencing this issue while using the mouse (so far, it hasn't occurred when using the tablet). I've tried all the graphics engines—DirectX, OpenGL, Vulkan—and the issue persists regardless. I've also tested with and without enabling the high-precision mouse option, and even disconnected my tablet in case there was any interference, but the problem still occurs. I'm sharing two videos where you can see that the mouse suddenly freezes.

The first one is more subtle:
[![Subtle freeze](https://img.youtube.com/vi/D_CHNN6BQrM/0.jpg)](https://youtu.be/D_CHNN6BQrM)

However, in the second one, it's quite obvious:
[![Significant freeze](https://img.youtube.com/vi/jZQKfVHFtYA/0.jpg)](https://youtu.be/jZQKfVHFtYA)

I should also mention that I haven't noticed any FPS drops or increases in ms during these occurrences.

If there is any additional information or specific tests you'd like me to run, please let me know. I'm happy to provide more details or assist in any way that could help diagnose and fix this issue.

**Comment by Trung0246** [comment]
For me this can be easily replicated if you switch between full screen and windowed mode on windows 11 multiple times, the cursor lag will getting more noticeable.

**Comment by BenCheung0422** [comment]
The poor performance on windowed mode would be expected as it is not as optimized as full screen mode. This includes rendering and also other window-related events (as well as input events). However the situation should be optimized, especially on full screen mode.

**Comment by AlejandroSuero** [comment]
@MEMORIEmusic I have the same issue since I started playing on macOS. I have a video (recorded a replay with OBS, to not affect performance) in this discussion https://github.com/ppy/osu/discussions/29782.

Tried with playing borderless, fullscreen and windowed, all to the same effect.

**Comment by AlejandroSuero** [comment]
I was playing a bit now while warming up and got 2 replays in the same map recorded from when it happens, different places and different amounts of times it happens:

**First try** (happened once):

<details>
<summary>video</summary>

https://github.com/user-attachments/assets/970cf607-2c8b-48d6-aaeb-6cb39466f7b8

</details>

**Second try** (happened twice:

<details>
<summary>videos</summary>

https://github.com/user-attachments/assets/493ff558-47b0-4f89-abb8-4cd022470ea5

https://github.com/user-attachments/assets/3a0b7fe4-8039-42eb-8be1-28e34b00f01a

</details>

[compressed-logs.zip](https://github.com/user-attachments/files/16946479/compressed-logs.zip)

**Comment by Apple2007** [comment]
I have also been facing this issue on the macOS version. It happens practically, and it usually kills my SS runs. I have tried everything suggested in the thread, and nothing has worked.

**Comment by Awzekek** [comment]
It's happened around 15 mins ago, just ruined 300pp score, but that doesn't matter. That lag happening very often and play with it unreal. Lag just stops cursor at one moment and then nothing happening, just lag at one moment.

https://github.com/user-attachments/assets/76cfffa2-3707-4b4a-b12f-8992599ece15


https://github.com/user-attachments/assets/897baea8-d2c3-40aa-9b8b-fefb8ad0af1a


 I can say, lazer start lagging, when you are in voice in discord. When you exit from voice (you don't have to close discord), lazer stops lagging at all and mouse lag disappearing

**Comment by Someone999** [comment]
I encuntered the same issue.
This is my case.
#33182

**Comment by Luk45135** [comment]
I've been having this issue on Linux as well. (Just mentioning this because i've only read reports from windows and macos people here)

**Comment by KTrain5169** [comment]
I'm not sure if this is related, but using trackpad with high-precision mouse input seems to also trigger this bug. disabling the high-precision mode seems to fix it but I haven't tested enough

**Comment by Hel1umHE** [comment]
i also encountered the same issue in the newest release
and i haven't saw this issue in previous update

https://github.com/user-attachments/assets/aeaf0eea-e0e2-46ef-9b04-df5c5179b9b0

**Comment by Someone999** [comment]
Additionally, I think there are also some input time precision issues.
 In the screenshots below, you'll notice that the hit timing distribution in osu!stable appears more accurate or expected, while in osu!lazer, the results seem skewed or inconsistent for similar gameplay.
﻿
This suggests there might be a discrepancy in how input hit times are recorded or judged between versions.

In stable, osu! mania
![Image](https://github.com/user-attachments/assets/d35fd17c-d238-4ea7-bea3-2930bb3bf422)
In lazer, osu! mania
![Image](https://github.com/user-attachments/assets/06f11510-fc29-419b-a77e-21ae1d4d08bb)

**Comment by peppy** [comment]
This is just more examples of input handling not using the async pathway which SDL3 will fix. But we have had to roll that back multiple times due to issues.

I believe the last blocker was breaking tablet support for some 🤷

**Comment by Joehuu** [comment]
Note that SDL3 may not fix this based on recent report in https://github.com/ppy/osu/issues/34348. Needs more investigation.

**Comment by peppy** [comment]
OTD *is* handling input in an async/threaded manner, so it's definitely not the same issue as we're used to seeing:

<img width="1435" height="1087" alt="Image" src="https://github.com/user-attachments/assets/8cba874c-c249-48b6-b789-e09f7972311a" />

**Comment by Skrubbadubba** [comment]
Today I am encoutering this issue aswell.

https://github.com/user-attachments/assets/d5308422-bf44-4f8c-9fde-cdac1c85afe3

This has been a problem to and fro the past week or so. Yesterday, I didnt experience it at all, the day before that I got it alot. At that time, whenever the input dropped, my case fans ramped up briefly, this isn't happening today.

### Details

At first when the input drops, nothing else lags. When the input is regained, for maybe 100 milliseconds or so as the input is "catching up" to my position, I get some frame drops. In the video, it appears as if they are alot of frame drop, this isn't the case. Altough this varies, sometimes it's accompanied by alot of frame drops (graphics freezing in rare cases), most of the time it isn't.

This seems to only happens in multithreaded mode. When in singlethreaded, what I assume to be the same problem is instead causing the entire game to lag (kind of what it looks like in the video), not just the input. The degree to which singlethreaded vs multithreaded affects this seems to also vary, but this is the general behaviour.

This is dependent on my PCs mood. No doubt this is to do with windows, other processes or input devies. I can't find what though. Closing applications and processes seem to lessen the severity of the stutters but not remove them.

Worth noting that I do get stuttering in alot of other games, probably related to windows finding novel ways to use my pc without me asking. Through some voodo magic stable is able to negate it though.

### Logs around the time of recording
<details>
<summary>Runtime logs:</summary>

```logs
2025-12-09 21:24:48 [verbose]: Game-wide working beatmap updated to peppy, Tom94 - THIS IS MY PP (Decon082) [9001 pp]
2025-12-09 21:24:48 [verbose]: Invalidating working beatmap cache for peppy, Tom94 - THIS IS MY PP (Decon082) [9001 pp]
2025-12-09 21:24:48 [verbose]: Song select decided to ensurePlayingSelected
2025-12-09 21:24:51 [verbose]: Focus contention triggered by UserModSelectOverlay.
2025-12-09 21:24:57 [verbose]: Game-wide working beatmap updated to peppy, Tom94 - THIS IS MY PP (Decon082) [0 pp]
2025-12-09 21:24:57 [verbose]: Invalidating working beatmap cache for peppy, Tom94 - THIS IS MY PP (Decon082) [0 pp]
2025-12-09 21:25:00 [verbose]: Invalidating working beatmap cache for peppy, Tom94 - THIS IS MY PP (Decon082) [0 pp]
2025-12-09 21:25:00 [verbose]: Game-wide working beatmap updated to peppy, Tom94 - THIS IS MY PP (Decon082) [0 pp]
2025-12-09 21:25:00 [verbose]: Invalidating working beatmap cache for peppy, Tom94 - THIS IS MY PP (Decon082) [0 pp]
2025-12-09 21:25:00 [verbose]: 📺 OsuScreenStack#179(depth:4) suspended SoloSongSelect#468 (waiting on SoloSongSelect+PlayerLoader#150)
2025-12-09 21:25:00 [verbose]: 📺 OsuScreenStack#179(depth:5) loading SoloSongSelect+PlayerLoader#150
2025-12-09 21:25:00 [verbose]: 📺 OsuScreenStack#179(depth:5) entered SoloSongSelect+PlayerLoader#150
2025-12-09 21:25:00 [verbose]: 🖍️ Shader Texture2D/Path loaded from cache!
2025-12-09 21:25:00 [verbose]: 🖍️ Shader PathPrepass/PathPrepass loaded from cache!
2025-12-09 21:25:01 [verbose]: Starting high performance session
2025-12-09 21:25:02 [verbose]: 📺 OsuScreenStack#179(depth:5) suspended SoloSongSelect+PlayerLoader#150 (waiting on SoloPlayer#464)
2025-12-09 21:25:02 [verbose]: 📺 OsuScreenStack#179(depth:6) entered SoloPlayer#464
2025-12-09 21:25:02 [verbose]: GameplayClockContainer seeking to 0
2025-12-09 21:25:02 [verbose]: GameplayClockContainer started via call to StartGameplayClock
2025-12-09 21:25:13 [verbose]: Received beatmap updates 1 updates with last id 9647509
2025-12-09 21:25:23 [verbose]: No token, skipping score submission
2025-12-09 21:25:23 [verbose]: No token, skipping score submission
2025-12-09 21:25:23 [verbose]: 📺 OsuScreenStack#179(depth:5) exit from SoloPlayer#464
2025-12-09 21:25:23 [verbose]: 📺 OsuScreenStack#179(depth:5) resume to SoloSongSelect+PlayerLoader#150
2025-12-09 21:25:23 [verbose]: Ending high performance session
2025-12-09 21:25:23 [verbose]: 🖍️ Shader Texture2D/Path loaded from cache!
2025-12-09 21:25:23 [verbose]: 🖍️ Shader PathPrepass/PathPrepass loaded from cache!
2025-12-09 21:25:23 [verbose]: Starting high performance session
2025-12-09 21:25:23 [verbose]: 📺 OsuScreenStack#179(depth:5) suspended SoloSongSelect+PlayerLoader#150 (waiting on SoloPlayer#275)
2025-12-09 21:25:23 [verbose]: 📺 OsuScreenStack#179(depth:6) entered SoloPlayer#275
2025-12-09 21:25:23 [verbose]: GameplayClockContainer seeking to 0
2025-12-09 21:25:24 [verbose]: GameplayClockContainer started via call to StartGameplayClock
2025-12-09 21:25:44 [verbose]: No token, skipping score submission
2025-12-09 21:25:45 [verbose]: Ending high performance session
2025-12-09 21:25:45 [verbose]: 📺 OsuScreenStack#179(depth:6) suspended SoloPlayer#275 (waiting on SoloResultsScreen#860)
2025-12-09 21:25:45 [verbose]: 📺 OsuScreenStack#179(depth:7) loading SoloResultsScreen#860
2025-12-09 21:25:45 [verbose]: 📺 OsuScreenStack#179(depth:7) entered SoloResultsScreen#860
2025-12-09 21:26:03 [verbose]: Received beatmap updates 1 updates with last id 9647510
2025-12-09 21:26:13 [verbose]: Received beatmap updates 1 updates with last id 9647511
2025-12-09 21:27:08 [verbose]: Received beatmap updates 2 updates with last id 9647513
2025-12-09 21:27:14 [verbose]: 📺 OsuScreenStack#179(depth:6) exit from SoloResultsScreen#860
2025-12-09 21:27:14 [verbose]: 📺 OsuScreenStack#179(depth:6) resume to SoloPlayer#275
2025-12-09 21:27:14 [verbose]: No token, skipping score submission
2025-12-09 21:27:14 [verbose]: 📺 OsuScreenStack#179(depth:5) exit from SoloPlayer#275
2025-12-09 21:27:14 [verbose]: 📺 OsuScreenStack#179(depth:5) resume to SoloSongSelect+PlayerLoader#150
2025-12-09 21:27:14 [verbose]: 🖍️ Shader Texture2D/Path loaded from cache!
2025-12-09 21:27:14 [verbose]: 🖍️ Shader PathPrepass/PathPrepass loaded from cache!
2025-12-09 21:27:14 [verbose]: Starting high performance session
2025-12-09 21:27:14 [verbose]: 📺 OsuScreenStack#179(depth:5) suspended SoloSongSelect+PlayerLoader#150 (waiting on SoloPlayer#389)
2025-12-09 21:27:14 [verbose]: 📺 OsuScreenStack#179(depth:6) entered SoloPlayer#389
2025-12-09 21:27:14 [verbose]: GameplayClockContainer seeking to 0
2025-12-09 21:27:14 [verbose]: GameplayClockContainer started via call to StartGameplayClock
2025-12-09 21:27:24 [verbose]: No token, skipping score submission
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:24 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:25 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:26 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: Frames arrived at SpectatorClient outside of gameplay scope and will be ignored.
2025-12-09 21:27:27 [verbose]: GameplayClockContainer stopped via call to StopGameplayClock
2025-12-09 21:27:37 [verbose]: No token, skipping score submission
2025-12-09 21:27:37 [verbose]: 📺 OsuScreenStack#179(depth:5) exit from SoloPlayer#389
2025-12-09 21:27:37 [verbose]: 📺 OsuScreenStack#179(depth:5) resume to SoloSongSelect+PlayerLoader#150
2025-12-09 21:27:37 [verbose]: Ending high performance session
2025-12-09 21:27:37 [verbose]: 🖍️ Shader Texture2D/Path loaded from cache!
2025-12-09 21:27:37 [verbose]: 🖍️ Shader PathPrepass/PathPrepass loaded from cache!
2025-12-09 21:27:37 [verbose]: Starting high performance session
2025-12-09 21:27:37 [verbose]: 📺 OsuScreenStack#179(depth:5) suspended SoloSongSelect+PlayerLoader#150 (waiting on SoloPlayer#341)
2025-12-09 21:27:37 [verbose]: 📺 OsuScreenStack#179(depth:6) entered SoloPlayer#341
2025-12-09 21:27:37 [verbose]: GameplayClockContainer seeking to 0
2025-12-09 21:27:37 [verbose]: GameplayClockContainer started via call to StartGameplayClock
2025-12-09 21:27:44 [verbose]: GameplayClockContainer stopped via call to StopGameplayClock
2025-12-09 21:28:08 [verbose]: Received beatmap updates 1 updates with last id 9647514
2025-12-09 21:28:47 [verbose]: No token, skipping score submission
2025-12-09 21:28:47 [verbose]: 📺 OsuScreenStack#179(depth:5) exit from SoloPlayer#341
2025-12-09 21:28:47 [verbose]: 📺 OsuScreenStack#179(depth:5) resume to SoloSongSelect+PlayerLoader#150
2025-12-09 21:28:47 [verbose]: Ending high performance session
2025-12-09 21:28:47 [verbose]: 📺 OsuScreenStack#179(depth:4) exit from SoloSongSelect+PlayerLoader#150
2025-12-09 21:28:47 [verbose]: 📺 OsuScreenStack#179(depth:4) resume to SoloSongSelect#468
2025-12-09 21:28:47 [verbose]: Invalidating working beatmap cache for peppy, Tom94 - THIS IS MY PP (Decon082) [0 pp]
2025-12-09 21:28:47 [verbose]: Carousel[op 11348172] 0 ms: Filter operation queued, waiting for 100 ms debounce
2025-12-09 21:28:47 [verbose]: Carousel[op 11348172] 101 ms: Performing BeatmapCarouselFilterMatching
2025-12-09 21:28:47 [verbose]: Carousel[op 11348172] 113 ms: Performing BeatmapCarouselFilterSorting
2025-12-09 21:28:47 [verbose]: Carousel[op 11348172] 113 ms: Performing BeatmapCarouselFilterGrouping
2025-12-09 21:28:47 [verbose]: Carousel[op 11348172] 113 ms: Updating Y positions
2025-12-09 21:28:47 [verbose]: Carousel[op 11348172] 114 ms: Items ready for display
2025-12-09 21:28:48 [verbose]: Carousel[op 61825962] 0 ms: Filter operation queued, waiting for 100 ms debounce
2025-12-09 21:28:48 [verbose]: Carousel[op 61825962] 101 ms: Performing BeatmapCarouselFilterMatching
2025-12-09 21:28:48 [verbose]: Carousel[op 61825962] 107 ms: Performing BeatmapCarouselFilterSorting
2025-12-09 21:28:48 [verbose]: Carousel[op 61825962] 161 ms: Performing BeatmapCarouselFilterGrouping
2025-12-09 21:28:48 [verbose]: Carousel[op 61825962] 171 ms: Updating Y positions
2025-12-09 21:28:48 [verbose]: Carousel[op 61825962] 174 ms: Items ready for display
2025-12-09 21:28:48 [verbose]: Invalidating working beatmap cache for peppy, Tom94 - THIS IS MY PP (Decon082) [0 pp]
2025-12-09 21:28:50 [verbose]: Focus contention triggered by NowPlayingOverlay.
2025-12-09 21:28:53 [verbose]: Focus contention triggered by SettingsOverlay.
2025-12-09 21:28:55 [verbose]: Scrolling to GameplaySection..
2025-12-09 21:28:56 [verbose]: Finished scrolling to GameplaySection!
2025-12-09 21:28:56 [verbose]: Scrolling to GeneralSection..
2025-12-09 21:28:57 [verbose]: Finished scrolling to GeneralSection!
2025-12-09 21:29:48 [verbose]: Received beatmap updates 1 updates with last id 9647515
2025-12-09 21:30:23 [verbose]: Received beatmap updates 1 updates with last id 9647516
```
</details>

<details>
<summary>Network logs:</summary>

```logs
2025-12-09 21:25:46 [verbose]: Performing request osu.Game.Online.API.Requests.GetBeatmapSetRequest
2025-12-09 21:25:46 [verbose]: Request to https://osu.ppy.sh/api/v2/beatmapsets/215968 failed with System.Net.WebException: NotFound.
2025-12-09 21:25:46 [verbose]: Response was: {"error":"Specificerad beatmap  kunde inte hittas."}
2025-12-09 21:25:46 [verbose]: Failing request osu.Game.Online.API.Requests.GetBeatmapSetRequest (osu.Game.Online.API.APIException: Specificerad beatmap  kunde inte hittas.
2025-12-09 21:25:46 [verbose]: ---> System.Net.WebException: NotFound
2025-12-09 21:25:46 [verbose]: --- End of inner exception stack trace ---)
2025-12-09 21:25:46 [verbose]: Favourite button failed to fetch beatmap info: osu.Game.Online.API.APIException: Specificerad beatmap  kunde inte hittas.
2025-12-09 21:25:46 [verbose]: ---> System.Net.WebException: NotFound
2025-12-09 21:25:46 [verbose]: at osu.Framework.IO.Network.WebRequest.Complete(Exception e)
2025-12-09 21:25:46 [verbose]: at osu.Framework.IO.Network.WebRequest.beginResponse(CancellationToken cancellationToken)
2025-12-09 21:25:46 [verbose]: at osu.Framework.IO.Network.WebRequest.internalPerform(CancellationToken cancellationToken)
2025-12-09 21:25:46 [verbose]: --- End of inner exception stack trace ---
2025-12-09 21:25:47 [verbose]: Performing request osu.Game.Online.API.Requests.ChatAckRequest
2025-12-09 21:25:47 [verbose]: Request to https://osu.ppy.sh/api/v2/chat/ack successfully completed!
2025-12-09 21:25:47 [verbose]: ChatAckRequest finished with response size of 15 bytes
2025-12-09 21:26:27 [verbose]: Request to https://auth.ppy.sh/updates/get successfully completed!
2025-12-09 21:26:47 [verbose]: Performing request osu.Game.Online.API.Requests.ChatAckRequest
2025-12-09 21:26:48 [verbose]: Request to https://osu.ppy.sh/api/v2/chat/ack successfully completed!
2025-12-09 21:26:48 [verbose]: ChatAckRequest finished with response size of 15 bytes
2025-12-09 21:27:28 [verbose]: Request to https://auth.ppy.sh/updates/get successfully completed!
2025-12-09 21:27:48 [verbose]: Performing request osu.Game.Online.API.Requests.ChatAckRequest
2025-12-09 21:27:48 [verbose]: Request to https://osu.ppy.sh/api/v2/chat/ack successfully completed!
2025-12-09 21:27:48 [verbose]: ChatAckRequest finished with response size of 15 bytes
2025-12-09 21:28:28 [verbose]: Request to https://auth.ppy.sh/updates/get successfully completed!
2025-12-09 21:28:47 [verbose]: Performing request osu.Game.Online.API.Requests.GetBeatmapSetRequest
2025-12-09 21:28:48 [verbose]: Request to https://osu.ppy.sh/api/v2/beatmapsets/215968 failed with System.Net.WebException: NotFound.
2025-12-09 21:28:48 [verbose]: Response was: {"error":"Specificerad beatmap  kunde inte hittas."}
2025-12-09 21:28:48 [verbose]: Failing request osu.Game.Online.API.Requests.GetBeatmapSetRequest (osu.Game.Online.API.APIException: Specificerad beatmap  kunde inte hittas.
```
</details>


I did not find any other logs that looked worth sharing.

### Edit:

This particular day, singelthreaded seems to resolve it completely (aside for small stutters which I assume is unrelated).

Toggling high precision mouse input has no effect.

Disconnecting my wired mouse has no effect.

Disconnecting my bluetooth adapted (some peripherals were connected) has no effect.

**Comment by peppy** [comment]
That's not input dropping that's your whole system struggling.

recommend diagnosing at a windows level.

**Comment by SevenSec114** [comment]
Here I want to supplement more details about what I encountered, hope this would not be too late.

Stuttering occurs only (exactly for me) at **specifically regular timing points** during beatmaps when **keyboard has input.** The following video shows that:
- The issue did not occur occasionally, rather occurred at some specific times within a beatmap
- At the same time if there is no click handling, it would not occur
  - simply without any keyboard input
  - `relax mod` enabled

https://github.com/user-attachments/assets/43e39aa4-d623-4790-85be-8ca5d962ca29

Also note that this issue came to me after version `osulazer-2025.912.0-lazer-linux-x64-full`(not included).

*I haven't found any log that is helpful but providing these experiences XD*

**Comment by realooffyy** [comment]
I have also run into this issue today, with both my tapping and cursor inputs occasionally dropped. The clip includes input drops between the first triple the first stream.
- Kernel: Linux 6.8.0-90-lowlatency
- 2026.108.0-tachyon

https://github.com/user-attachments/assets/e87f5a90-0a50-4a49-9fb9-8680c457eaa8

**Comment by pviscone** [comment]
Hello, I noticed the same effect today for the first time (on linux).
I was observing constant and regular drops of the input thread and i started tracing system calls of the osu main thread.

```
> sudo strace -tp 278273 -T -e poll,futex,read

...
04:22:46 poll([{fd=189, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout) <0.000017>
04:22:46 poll([{fd=177, events=POLLIN}], 1, 0) = 0 (Timeout) <0.000016>
04:22:46 poll([{fd=189, events=POLLIN|POLLPRI}], 1, 0) = 0 (Timeout) <0.000021>
04:22:46 poll([{fd=177, events=POLLIN}], 1, 0) = 1 ([{fd=177, revents=POLLIN}]) <0.000011>
04:22:46 read(176, "DEVTYPE=power_supply\nPOWER_SUPPL"..., 4104) = 615 <0.000759> 
```

It seems that it was constantly checking the power status of my machine.
Indeed, my laptop was on charge but the battery was not 100% yet. As soon as the battery reached full charge osu stopped checking the power status and the cursor issue disappeared.

I don't know how many people here that are affected by this issue were in these exact conditions (on a not fully charged laptop, even if on charge)

**Comment by peppy** [comment]
> I was observing constant and regular drops of the input thread and i started tracing system calls of the osu main thread.

This sounds super weird. We never poll for the battery level. The only time we check is once at the beatmap loading screen:

https://github.com/ppy/osu/blob/3ae98fd6649c72e561dcdde498ecd519c75f35d8/osu.Game/Screens/Play/PlayerLoader.cs#L805-L817

Is this where you were seeing the stutter, or during gameplay?

**Comment by pviscone** [comment]
> Is this where you were seeing the stutter, or during gameplay?

I was being affected by the stutter everywhere, both in the menu and during gameplay

**Comment by pviscone** [comment]
even more weirdly, I just noticed that the issue is present only if the battery is not fully charged AND the laptop is charging. If you unplug the laptop the problem disappear, also if the battery is not fully charged.

Everything happen in menu, during gameplay, everywhere in real-time, i.e. if i unplug the charger it stops stuttering, and when plugged in again (with battery at <100%) it restart stuttering, all of this without restarting the game

**Comment by kuschanow** [comment]
I have the same problem, but I can't record it because it happens unpredictably. It can happen once, and then I can play normally for three days. I use a Wacom CTL-472. I don't notice this issue on my old Wacom MTE-450. Also, I don't play on a laptop. Actually, my computer has more than enough resources for a game like osu.

I play on Windows

**Comment by KomoreLeaf** [comment]
Hello! I am using Gaomon S620 with the original dirver v14.8.133.1259 ver
I come from Stable using mouse mode in tablet driver and 1.6 sensitivity in osu mouse settings with raw input on
When i put the same settings in laser it doesn't work like in stable with high precision input mouse it causes the same issue as topic starter but only when i drag when i hover it doesnt teleporting but i cant play hover that way.
If i turning off high precision input mouse it works well but not my settings anymore

https://youtu.be/fARa6g9T8Sg

**Comment by peppy** [comment]
This should be fixed with recent changes, please report back if you're still having issues.

**Comment by swipy2** [comment]
hello this issue is still happening to me. I tried everything. this only happens when i use the osu lazer client, this doesn't happen on stable.
i tried using tachyon, windowed fullscreen and normal fullscreen, high precision mouse on and off. i basically tried everything. Does anyone have a fix for this please!
